### PR TITLE
improve handling of test output

### PR DESCRIPTION
**Pull request by tomerd**
motivation: fix cases where test output is formatted incorrectly due to excessive newline addition

changes: preserve output from test hartness as-is without attempting to remove oradd newlines

### CODE REVIEW

**Comment by tomerd** [comment]
fixes https://github.com/apple/swift-package-manager/issues/6566

**Comment by tomerd** [comment]
@swift-ci test

**Review by MaxDesiatov — APPROVED** [review]

**Review by neonichu — APPROVED** [review]

**Comment by grynspan** [comment]
I had a PR ready to go but you beat me to it :)

File: Sources/Commands/SwiftTestTool.swift [hunk 1]
```diff
@@ -390,7 +390,7 @@ public struct SwiftTestTool: SwiftCommand {
         let ranSuccessfully = runner.test(outputHandler: {
             // command's result output goes on stdout
             // ie "swift test" should output to stdout
-            print($0)
+            print($0, terminator: "")
```
Comment (grynspan) [inline Sources/Commands/SwiftTestTool.swift:393]: There are some other necessary changes that are missing. Look for other uses of `test(outputHandler` in the repo.

**Review by grynspan — CHANGES_REQUESTED** [review]
Some missing bits.

**Comment by tomerd** [comment]
@swift-ci test

File: Sources/Commands/SwiftTestTool.swift (hunk 1 above)
Comment (tomerd) [reply Sources/Commands/SwiftTestTool.swift:393]: https://github.com/apple/swift-package-manager/pull/7126/commits/37ec933ec6cd8daea9849567d9512a1b2ea21e3a

Comment (tomerd) [reply Sources/Commands/SwiftTestTool.swift:393]: @grynspan ^^

Comment (grynspan) [reply Sources/Commands/SwiftTestTool.swift:393]: Woohoo!

**Review by grynspan — APPROVED** [review]

**Comment by tomerd** [comment]
@swift-ci test windows

**Pull request merged**